src/main/java/net/masterthought/cucumber/json/Step.java

```diff
@@ -13,7 +15,9 @@
     private final Row[] rows = new Row[0];
     private final Match match = null;
     private final Embedding[] embeddings = new Embedding[0];
-    private final Output output = null;
+    @JsonDeserialize(using = OutputsDeserializer.class)
+    @JsonProperty("output")
+    private final Output[] outputs = new Output[0];
```

Comment:
Have you tried to use single class and deserialilze it inside? so line 19 won't be required and deserializer method will be linked iwth Outputs class. Then also there will be single attribute here Objects, not the array

So it is more like the current solution